### ISSUE TITLE
core.rotate_node change breaks many mods

### ISSUE THREAD

**Issue by Megaf** [minetest/minetest#6899]
PR: #6648

That seems to be breaking many mods.

Mod Issue: https://github.com/minetest-mods/nixie_tubes/issues/8

Comment on PR: https://github.com/minetest/minetest/pull/6648#issuecomment-356130299

**Comment by SmallJoker** [comment]
@VanessaE @Megaf 
I believe the best solution for this problem is to adjust `param2` for existing nodes in `core.item_place_node` and only run the callbacks for newly placed nodes. But that would also require a change in nixie_tubes to not remove the node, which again would end up in an endless loop.
If you have a better idea to solve the callback loop but still call `register_on_placenode()` for rotated nodes `on_place`.

**Comment by VanessaE** [comment]
It worked perfectly when I submitted the original pull request.  If others' rewrites of it since then broke it, well, I can't do much about that :stuck_out_tongue_closed_eyes:

I thought it was documented just fine:  https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L3376

In retrospect, I should have mentioned in there about calling it from on-/after-place, but since the code from which it was derived had been in use for months already, I guess it seemed obvious.

It wouldn't just be nixies, it would be all mods that use the rotation function the way I had intended:  `rotate_node` is referenced 91 times in my own mods, 139 times across my entire mods library, and 6 times in minetest_game (where it's used properly).

Are you expecting to trigger on-/after-place callbacks in response to `set_node()` ?  If so, that's wrong:  https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L4764-L4766  ...  that's the only way I can see a loop happening here.
